solidity-v1/dashboard/src/hooks/useAutoConnect.js

```diff
@@ -42,59 +42,38 @@ const useAutoConnect = () => {
     [walletAddressFromUrl]
   )
 
+  const connectWallet = (connector) => {
+    connector.getAccounts().then((accounts) => {
+      setInjectedTried(true)
+      if (
+        (!isEmptyArray(accounts) && isExactRoutePath) ||
+        isWalletFromUrlSameAsInMetamask(accounts)
+      ) {
+        connectAppWithWallet(connector, false).catch((error) => {
+          // Just log an error, we don't want to do anything else.
+          console.log(
+            "Eager injected connector cannot connect with the dapp:",
+            error.message
+          )
+        })
+      } else if (walletAddressFromUrl && !connector) {
+        const explorerModeConnector = new ExplorerModeConnector()
+        openModal(MODAL_TYPES.ExplorerMode, {
+          connectAppWithWallet,
+          connector: explorerModeConnector,
+          address: walletAddressFromUrl,
+          connectWithWalletOnMount: true,
+        })
+      }
+    })
+  }
```

Comment:
I think we should move this function inside the `useEffect()` function or refactor it to `useCallback()` with proper dependencies. I think the first approach would be the best because we are not using this function outside of the `useEeffect`

For more infor please see [Can I call separate function in useEffect?](https://stackoverflow.com/a/61918728) and [Define a function inside useEffect or outside?](https://stackoverflow.com/a/61528022)